src/game/mod.rs

```diff
@@ -0,0 +1,227 @@
+use std::collections::HashMap;
+
+use crate::conways_error::ConwaysError;
+use crate::cell::Cell;
+
+pub struct Game {
+    live_cells: Vec<Cell>,
+}
+
+impl Game {
+    pub fn new_with_seed(seed: Vec<(i32,i32)>) -> Self {
+        let live_cells = seed.iter().map(|(x,y)| Cell::new(*x,*y)).collect();
+
+        Game {
+            live_cells
+        }
+    }
+
+    pub fn new() -> Self {
+        let live_cells = vec![
+            Cell::new(1, 3),
+            Cell::new(1, 2),
+            Cell::new(1, 1),
+        ];
+
+        Game {
+            live_cells
+        }
+    }
+
+    // Returns the positions of the 8 neighbours of a cell
+    fn neighbour_positions(&self, cell: &Cell) -> Vec<Cell> {
+        let mut positions = Vec::with_capacity(8);
+
+        for dx in -1..=1 {
+            for dy in -1..=1 {
+                if dx == 0 && dy == 0 {
+                    continue;
+                }
+
+                positions.push(Cell { x: cell.x + dx, y: cell.y + dy });
+            }
+        }
+
+        positions
+    }
+
+    // Create a new_live_live cells vec by populating it with the dead cells that have 3 live neighbours and the current live cells that have 2 or 3 live neighbours
+    pub fn new_live_cells(&self) -> Vec<Cell> {
+        let mut dead_cells_map = HashMap::new();
+        let mut neighbours_count = HashMap::new();
+
+        for cell in self.live_cells.iter() {
+            for neighbor in self.neighbour_positions(cell) {
+                if !self.live_cells.contains(&neighbor) {
+                    let counter = dead_cells_map.entry(neighbor).or_insert(0);
+                    *counter += 1;
+                } else {
+                    let counter = neighbours_count.entry(cell.clone()).or_insert(0);
+                    *counter += 1;
+                }
+            }
+        }
+
+        let mut new_live_cells = Vec::new();
+        for (dead_cell, count) in dead_cells_map {
+            if count == 3 {
+                new_live_cells.push(dead_cell);
+            }
+        }
+
+        for (cell, count) in neighbours_count {
+            if count == 2 || count == 3 {
+                new_live_cells.push(cell);
+            }
+        }
+
+        new_live_cells
+    }
+
+    pub fn update_game(&mut self) -> Result<(), ConwaysError> {
```

Comment:
Maybe is a little rushed making `update_game` return a `Result`, as we don't really returns anything and is making the tests doing the `let _ = game.update_game()`

Reply:
Done!